File: clients/health-android/src/main/kotlin/io/zen70/healthgateway/GatewayIdentityContext.kt
package io.zen70.healthgateway

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * JWT identity context decoded from the gateway-issued access token.
 *
 * Claim names match the backend JWT encoder (backend/core/auth.py):
 * sub, tenant_id, scopes, iat, exp.
 *
 * [expiresAt] is a Unix epoch second (Long) to match the JWT `exp` claim type.
 * Callers must check [expiresAt] before using the context and trigger token
 * refresh via the gateway /v1/auth/refresh endpoint when within 60 seconds
 * of expiry.
 */
@Serializable
data class GatewayIdentityContext(
    @SerialName("sub") val subject: String,
    @SerialName("tenant_id") val tenantId: String,
    @SerialName("scopes") val scopes: List<String>,
    @SerialName("iat") val issuedAt: Long,
    @SerialName("exp") val expiresAt: Long,
)

File: clients/health-ios/Sources/HealthGatewayClient/GatewayIdentityContext.swift
import Foundation

public struct GatewayIdentityContext: Sendable, Equatable, Codable {
    public let subject: String
    public let tenantID: String
    public let scopes: [String]
    public let issuedAt: Date
    public let expiresAt: Date

    public init(
        subject: String,
        tenantID: String,
        scopes: [String],
        issuedAt: Date,
        expiresAt: Date
    ) {
        self.subject = subject
        self.tenantID = tenantID
        self.scopes = scopes
        self.issuedAt = issuedAt
        self.expiresAt = expiresAt
    }

    // JWT claim names as emitted by the backend JWT encoder (backend/core/auth.py).
    enum CodingKeys: String, CodingKey {
        case subject = "sub"
        case tenantID = "tenant_id"
        case scopes
        case issuedAt = "iat"
        case expiresAt = "exp"
    }
}
